docs/modules/deployment/pages/upgrade/introduction.adoc

```diff
@@ -0,0 +1,64 @@
+
+[[upgrade]]
+= Upgrade {page-component-title}
+
+This section outlines the steps to follow to upgrade to a newer version of {page-component-title}.
+You may need to complete additional steps to upgrade in a more complex setup (running more than one OpenNMS instance, database migration requirements, age of current version, and so on).
+
+A basic upgrade with one instance takes about an hour.
+This time varies, and will be longer depending on how customized and complex your setup is.
+
+[[byb_upgrade]]
+== Before you begin
+
+There are a number of tasks you should complete before starting an upgrade:
+
+ifeval::["{page-component-title}" == "Meridian"]
+* Check the https://meridian.opennms.com/[release notes] for breaking changes in each release between your current version and the version you want to upgrade to.
+Address any issues that might affect you.
+endif::[]
+
+ifeval::["{page-component-title}" == "Horizon"]
+* Check the https://vault.opennms.com/docs/opennms/releases/index.html[release notes] for breaking changes in each release between your current version and the version you want to upgrade to.
+Address any issues that might affect you.
+
+NOTE: Release notes for versions 28 and 29 are not available at the vault docs link (above) at this time.
+Please check the branches on GitHub for those release notes.
+endif::[]
+
+* Stop OpenNMS and PostgreSQL
+* xref:operation:admin/housekeeping/introduction.adoc#prune-events[Prune unneeded events]
+* <<non-root-permissions, Update permissions>> (if upgrading from pre-29 Horizon or pre-2022 Meridian)
```

Comment:
This step should be done closer to the end of the upgrade process, after installing updated packages and reconciling config file changes but before starting services. The script to fix permissions isn't available until after 29/2022 packages are installed.

Reply:
Removed, and will add as part of upgrade process procedure when that is added.